app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt

```diff
@@ -178,27 +187,64 @@ class MainActivity : AppCompatActivity(),
                     val options = BitmapFactory.Options()
                     options.inDensity = DisplayMetrics.DENSITY_DEFAULT
                     bitmap = BitmapFactory.decodeStream(stream, null, options)
+
+                    insertImageAndSimulateUpload(bitmap, mediaPath)
+                }
+                REQUEST_MEDIA_CAMERA_VIDEO -> {
+                    mediaPath = data?.data.toString()
                 }
                 REQUEST_MEDIA_VIDEO -> {
+                    mediaPath = data?.data.toString()
+
+                    aztec.videoThumbnailGetter?.loadVideoThumbnail(mediaPath, object : Html.VideoThumbnailGetter.Callbacks {
+                        override fun onThumbnailFailed() {
+                        }
+
+                        override fun onThumbnailLoaded(drawable: Drawable?) {
+                            val conf = Bitmap.Config.ARGB_8888 // see other conf types
+                            bitmap = Bitmap.createBitmap(drawable!!.intrinsicWidth, drawable.intrinsicHeight, conf)
+                            val canvas = Canvas(bitmap)
+                            drawable.setBounds(0, 0, canvas.width, canvas.height)
+                            drawable.draw(canvas)
+
+                            insertVideoAndSimulateUpload(bitmap, mediaPath)
+                        }
+
+                        override fun onThumbnailLoading(drawable: Drawable?) {
+                        }
+
+                    }, this.resources.displayMetrics.widthPixels)
                 }
             }
-
-            insertMediaAndSimulateUpload(bitmap, mediaPath)
         }
 
         super.onActivityResult(requestCode, resultCode, data)
     }
 
-    fun insertMediaAndSimulateUpload(bitmap: Bitmap?, mediaPath: String) {
+    fun insertImageAndSimulateUpload(bitmap: Bitmap?, mediaPath: String) {
+        val (id, attrs) = generateAttributesForMedia(mediaPath)
+        val mediaSpan = aztec.insertImage(BitmapDrawable(resources, bitmap), attrs)
+        insertMediaAndSimulateUpload(id, attrs, mediaSpan)
+    }
+
+    fun insertVideoAndSimulateUpload(bitmap: Bitmap?, mediaPath: String) {
+        val (id, attrs) = generateAttributesForMedia(mediaPath)
+        val mediaSpan = aztec.insertVideo(BitmapDrawable(resources, bitmap), attrs)
+        insertMediaAndSimulateUpload(id, attrs, mediaSpan)
+    }
+
+    private fun generateAttributesForMedia(mediaPath: String): Pair<String, AztecAttributes> {
         val id = (Math.random() * Int.MAX_VALUE).toString()
 
         val attrs = AztecAttributes()
         attrs.setValue("src", mediaPath) // Temporary source value.  Replace with URL after uploaded.
         attrs.setValue("id", id)
         attrs.setValue("uploading", "true")
+        attrs.setValue("video", "true")
```

Comment:
This attribute will also be applied to images inserted through `insertImageAndSimulateUpload` and will result in "play" button overlay on top of them